From 8511110f234180bc034f605436c25ed38e0e7ebe Mon Sep 17 00:00:00 2001
From: "Ruan, Hongfu" <hongfu.ruan@intel.com>
Date: Wed, 29 Dec 2021 10:38:18 +0800
Subject: [PATCH] adjust system sepolicy to allow houdini changes

houdini module needs special access right that needs to be granted
here.

Tracked-On: OAM-100437
Signed-off-by: Ruan, Hongfu <hongfu.ruan@intel.com>
---
 private/property.te | 3 ++-
 public/property.te  | 2 +-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/private/property.te b/private/property.te
index 29f4f1a0f..206a0df3a 100644
--- a/private/property.te
+++ b/private/property.te
@@ -61,7 +61,7 @@ neverallow { domain -coredomain } {
   -system_public_property_type
 }:file no_rw_file_perms;
 
-neverallow { domain -coredomain } {
+neverallow { domain -coredomain -vendor_init} {
   system_property_type
   -system_public_property_type
 }:property_service set;
@@ -584,6 +584,7 @@ neverallow {
 neverallow {
   domain
   -init
+  -vendor_init
 } default_prop:property_service set;
 
 # Only one of system_property_type and vendor_property_type can be assigned.
diff --git a/public/property.te b/public/property.te
index 1d3f358fd..4f4d101b4 100644
--- a/public/property.te
+++ b/public/property.te
@@ -218,7 +218,7 @@ system_public_prop(wifi_prop)
 system_public_prop(zram_control_prop)
 
 # Properties which don't have entries on property_contexts
-system_internal_prop(default_prop)
+system_public_prop(default_prop)
 
 # Properties used in default HAL implementations
 vendor_internal_prop(rebootescrow_hal_prop)
-- 
2.29.2

